File: lib/utils/enums/agml_geometry_type_enum.dart

enum AGMLGeometryTypeEnum {
  point,
  polyline,
  polygon,
  multipoint,
}

extension AGMLGeometryTypeEnumExtension on AGMLGeometryTypeEnum {
  String getString() {
    switch(this) {
      case AGMLGeometryTypeEnum.point: return 'POINT';
      case AGMLGeometryTypeEnum.polyline: return 'POLYLINE';
      case AGMLGeometryTypeEnum.polygon: return 'POLYGON';
      case AGMLGeometryTypeEnum.multipoint: return 'MULTIPOINT';
      default: return 'POINT';
    }
  }

  static AGMLGeometryTypeEnum fromString(String value) {
    switch (value) {
      case 'POINT':
        return AGMLGeometryTypeEnum.point;
      case 'POLYLINE':
        return AGMLGeometryTypeEnum.polyline;
      case 'POLYGON':
        return AGMLGeometryTypeEnum.polygon;
      case 'MULTIPOINT':
        return AGMLGeometryTypeEnum.multipoint;
      default:
        throw ArgumentError('Invalid geometry type string: $value');
    }
  }
}

File: android/src/main/kotlin/com/ludycom/arcgis_maps/utils/AGMLGeometryTypeEnum.kt
package com.ludycom.arcgis_maps.utils

import com.arcgismaps.geometry.GeometryType

enum class AGMLGeometryTypeEnum {
    POINT {
        override fun getValue() : GeometryType {
            return GeometryType.Point
        }
    },
    POLYLINE {
        override fun getValue() : GeometryType {
            return GeometryType.Polyline
        }
    },
    POLYGON {
        override fun getValue() : GeometryType {
            return GeometryType.Polygon
        }
    },
    MULTIPOINT {
        override fun getValue() : GeometryType {
            return GeometryType.Multipoint
        }
    };

    abstract fun getValue(): GeometryType
}